src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -3,36 +3,11 @@
 import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
 
-    /**
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Track.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Machine> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
why in this method you used ? extends Machine and int fill ? super T ?

Reply:
I thought that I needed to set such restrictions here, but with T instead of Machine, it`s still working.